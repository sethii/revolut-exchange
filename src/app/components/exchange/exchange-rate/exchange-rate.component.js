import React from 'react';
import PropTypes from 'prop-types';
import { setPropTypes, mapProps, compose } from 'recompose';
import getSymbolFromCurrency from 'currency-symbol-map'
import { twoDecimalPlacesFormatter, fourDecimalPlacesFormatter} from '../../../utils';
import styled from 'react-emotion';

const enhance = compose(
  setPropTypes({
    from: PropTypes.string.isRequired,
    to: PropTypes.string.isRequired,
    rate: PropTypes.number.isRequired
  }),
  mapProps((props) => ({
    ...props,
    rate: twoDecimalPlacesFormatter(props.rate),
    lastTwoCharacters: fourDecimalPlacesFormatter(props.rate).substr(-2)
  }))
);

const Paragraph = styled.p`
  font-size: 16px;
  color: #048bf2;
  border-radius: 20px;
  border: 2px solid #efefef;
  position: absolute;
  padding: 5px 20px;
  top: calc(50% - 16px);
  left: calc(50% - 60px);
  margin: 0;
  
  & span {
    font-size: 12px
  }
`;

export const ExchangeRate = enhance(({ from, to, rate, lastTwoCharacters }) => <Paragraph>
  <span>{ getSymbolFromCurrency(from) }</span>1 = <span>{ getSymbolFromCurrency(to) }</span>{ rate }<span>{ lastTwoCharacters }</span>
</Paragraph>);
