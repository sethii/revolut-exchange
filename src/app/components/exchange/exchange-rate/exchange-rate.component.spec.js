import React from 'react';
import renderer from 'react-test-renderer';
import { toClass } from 'recompose';

import { ExchangeRate } from './exchange-rate.component';

describe('Exchange rate component', () => {
  it('renders exchange rate', () => {
    const ExchangeRateComponent = toClass(ExchangeRate);
    const from = 'USD';
    const to = 'EUR';
    const rate = 0.123345;

    expect(renderer.create(
      <ExchangeRateComponent from={from} to={to} rate={rate}/>
    ).toJSON()).toMatchSnapshot();
  });
});
