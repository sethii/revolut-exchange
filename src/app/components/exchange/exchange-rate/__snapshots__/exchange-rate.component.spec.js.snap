// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exchange rate component renders exchange rate 1`] = `
.glamor-1 {
  font-size: 16px;
  color: #048bf2;
  border-radius: 20px;
  border: 2px solid #efefef;
  position: absolute;
  padding: 5px 20px;
  top: calc(50% - 16px);
  left: calc(50% - 60px);
  margin: 0px;
}

.glamor-1 span {
  font-size: 12px;
}

<p
  className="glamor-0 glamor-1"
>
  <span>
    $
  </span>
  1 = 
  <span>
    â‚¬
  </span>
  0.12
  <span>
    33
  </span>
</p>
`;
